openvdb/io/Compression.cc

```diff
@@ -22,11 +21,12 @@ compressionToString(uint32_t flags)
 {
     if (flags == COMPRESS_NONE) return "none";
 
-    std::vector<std::string> words;
-    if (flags & COMPRESS_ZIP) words.push_back("zip");
-    if (flags & COMPRESS_BLOSC) words.push_back("blosc");
-    if (flags & COMPRESS_ACTIVE_MASK) words.push_back("active values");
-    return boost::join(words, " + ");
+    std::string descr;
+    if (flags & COMPRESS_ZIP) descr += "zip + ";
+    if (flags & COMPRESS_BLOSC) descr += "blosc + ";
+    if (flags & COMPRESS_ACTIVE_MASK) descr += "active values + ";
+    if (!descr.empty()) descr.resize(descr.size() - 3);
```

Comment:
This is pretty clunky. You can't have ZIP and BLOSC defined at the same time, so I think I'd prefer this to just be explicit for readability:

```
if (flags & COMPRESS_ZIP) {
    if (flags & COMPRESS_ACTIVE_MASK)    return "zip + active values";
    else                                 return "zip";
}
// etc
```

Reply:
Technically, as far as this function is concerned, blosc and zip flags can exist at the same time as they're different bit toggles i.e. compressionToString(7) == "zip + blosc + active values". I realise that it shouldn't be (isn't) possible for them to both to exist, but seems that this method works without that context

Reply:
Perhaps we error or assert that both are not supplied at once, I still prefer the explicit solution here for readability.